adafruit_requests.py

```diff
@@ -561,40 +578,48 @@ def _send_request(
         data: Any,
         json: Any,
     ):
-        # pylint: disable=too-many-arguments
-        self._send(socket, bytes(method, "utf-8"))
-        self._send(socket, b" /")
-        self._send(socket, bytes(path, "utf-8"))
-        self._send(socket, b" HTTP/1.1\r\n")
-        if "Host" not in headers:
-            self._send(socket, b"Host: ")
-            self._send(socket, bytes(host, "utf-8"))
-            self._send(socket, b"\r\n")
-        if "User-Agent" not in headers:
-            self._send(socket, b"User-Agent: Adafruit CircuitPython\r\n")
-        # Iterate over keys to avoid tuple alloc
-        for k in headers:
-            self._send(socket, k.encode())
-            self._send(socket, b": ")
-            self._send(socket, headers[k].encode())
-            self._send(socket, b"\r\n")
+        # Convert data
+        content_type_header = None
+
+        # If json is sent, set content type header and convert to string
         if json is not None:
             assert data is None
+            content_type_header = "application/json"
             data = json_module.dumps(json)
-            self._send(socket, b"Content-Type: application/json\r\n")
-        if data:
-            if isinstance(data, dict):
-                self._send(
-                    socket, b"Content-Type: application/x-www-form-urlencoded\r\n"
-                )
-                _post_data = ""
-                for k in data:
-                    _post_data = "{}&{}={}".format(_post_data, k, data[k])
-                data = _post_data[1:]
-            if isinstance(data, str):
-                data = bytes(data, "utf-8")
-            self._send(socket, b"Content-Length: %d\r\n" % len(data))
+
+        # If data is sent and it's a dict, set content type header and convert to string
+        if data and isinstance(data, dict):
+            content_type_header = "application/x-www-form-urlencoded"
+            _post_data = ""
+            for k in data:
+                _post_data = "{}&{}={}".format(_post_data, k, data[k])
+            # remove first "&" from concatenation
+            data = _post_data[1:]
+
+        # Convert str data to bytes
+        if data and isinstance(data, str):
+            data = bytes(data, "utf-8")
+
+        self._send_as_bytes(socket, method)
+        self._send_as_bytes(socket, " /")
```

Comment:
```suggestion
        self._send(socket, b" /")
```

Reply:
fixed in https://github.com/adafruit/Adafruit_CircuitPython_Requests/pull/149/commits/5fa5aa218bbaecdd7b81c1e33a0894de635a8eff